### PR TITLE
Fix to invoke names in LFRic PSyIR algorithm code transformation (closes 2214)

### PR DESCRIPTION
**Pull request by rupertford**
The new PSyIR-based LFRic algorithm code transformation was producing invoke names that did not match the psy-layer subroutine names when there is one kernel and that kernel is a builtin. This PR fixes this error.

### CODE REVIEW

**Comment by rupertford** [comment]
This is ready for review from anyone. CI is giving a link error but I've not changed any code associated with links and the supposedly broken link works if I cut and paste it into my browser.

File: src/psyclone/tests/domain/lfric/algorithm/psyir/lfric_alg_invoke_call_test.py [hunk 1]
```diff
@@ -145,3 +145,30 @@ def test_aic_defcontainerrootname():
     routine_node = psyir.children[0]
     name = invoke._def_container_root_name(routine_node)
     assert name == "alg1_psy"
+
+
+def test_aic_defroutinerootname():
+    '''Check that _def_container_root_name returns the expected
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/algorithm/psyir/lfric_alg_invoke_call_test.py:151]: s/container/routine/

File: src/psyclone/tests/domain/lfric/algorithm/psyir/lfric_alg_invoke_call_test.py [hunk 2]
```diff
@@ -145,3 +145,30 @@ def test_aic_defcontainerrootname():
...
+        "  type(field_type) :: field1\n"
+        "  call invoke(kern(field1))\n"
+        "end subroutine alg1\n")
+    psyir = create_alg_psyir(code)
+    invoke = psyir.children[0][0]
+    assert isinstance(invoke, LFRicAlgorithmInvokeCall)
+    name = invoke._def_routine_root_name()
+    assert name == "invoke_0_kern"
+    # Single builtin should not have the name of the builtin added
+    builtin_code = code.replace("kern(field1)", "setval_c(field1, 0.0)")
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/algorithm/psyir/lfric_alg_invoke_call_test.py:169]: Up to you but you could parametrize (sic)  this test.

File: src/psyclone/domain/lfric/algorithm/psyir/lfric_alg_invoke_call.py [hunk 3]
```diff
@@ -68,6 +68,21 @@ def _def_container_root_name(node):
...
+        :rtype: str
+
+        '''
+        name = super()._def_routine_root_name()
+        if (len(self.children) == 1 and
+                isinstance(self.children[0], LFRicBuiltinFunctor)):
+            # By default the name of the kernel is added if there is
+            # only one functor. However we don't add this in LFRic if
+            # the functor is a builtin.
+            name = f"invoke_{self._index}"
```
Comment (arporter) [inline src/psyclone/domain/lfric/algorithm/psyir/lfric_alg_invoke_call.py:83]: perhaps:

        return f"invoke_{self._index}"
    else:
        return super()._def_routine_root_name()

**Review by arporter — CHANGES_REQUESTED** [review]
Just a few small things to tweak. (I would have let it through but I couldn't bring myself to ignore one of them.) I've kicked-off the integration tests and will check the results next time.

**Comment by arporter** [comment]
Just a note to say that the integration tests were all green.

File: src/psyclone/tests/domain/lfric/algorithm/psyir/lfric_alg_invoke_call_test.py (hunk 2 above)
Comment (rupertford) [reply src/psyclone/tests/domain/lfric/algorithm/psyir/lfric_alg_invoke_call_test.py:169]: OK, done.

File: src/psyclone/domain/lfric/algorithm/psyir/lfric_alg_invoke_call.py (hunk 3 above)
Comment (rupertford) [reply src/psyclone/domain/lfric/algorithm/psyir/lfric_alg_invoke_call.py:83]: Modified (but without the 'else')

**Comment by rupertford** [comment]
Ready for next review from @arporter

**Pull request merged**